Loader/SymfonyFixturesLoader.php

```diff
@@ -72,6 +79,48 @@ protected function createFixture($class)
         return $this->loadedFixtures[$class];
     }
 
+    /**
+     * Returns the array of data fixtures to execute.
+     *
+     * @param string $set
+     *
+     * @return array $fixtures
+     */
+    public function getFixtures($set = null)
+    {
+        $fixtures = parent::getFixtures();
+
+        if ($set) {
+            $filteredFixtures = [];
+            foreach ($fixtures as $class => $fixture) {
+                if (isset($this->setsFixtureMapping[$set][get_class($fixture)])) {
```

Comment:
You should use the variable `$class` here instead of calling `get_class($fixture)`

Reply:
my tests fail if i change this. if i test this it works. any idea why ?. i will try to fix so i can run the tests local

Reply:
I see. If one of the fixtures implements `DependentFixtureInterface`, the returned array of fixtures will not have the fixture classes as keys.

This is an unfortunate irregularity of the existing symfony code. The error is in Doctrine/Common/DataFixtures/Loader.php:308

Reply:
hm i see, best to leave it like this then ?

Reply:
Yes, I guess so.

Reply:
I think you should change the loop, so that you do not assign the array key to the variable `$class`. That might confuse other developers so they make the same mistake.

If you want to keep the variable `$class`, you should assign to it on the first line of the loop:
```
    $class = get_class($fixture);
```